virttest/test_setup.py

```diff
@@ -799,7 +799,13 @@ def __init__(self, params=None):
             if u_port not in ports:
                 ports.append(u_port)
             self.iptables_rules = self._assemble_iptables_rules(ports)
-            self.physical_nic = params.get("physical_nic")
+            physical_nic = params.get("physical_nic")
+            cmd_result = process.run(physical_nic, shell=True,
+                                     ignore_status=True)
+            if cmd_result.exit_status == 0:
+                self.physical_nic = cmd_result.stdout_text
+            else:
+                self.physical_nic = physical_nic
```

Comment:
Code can be optimized to:
```
self.physical_nic = params.get("physical_nic")
cmd_result = process.run(self.physical_nic, shell=True, ignore_status=True)
if cmd_result.exit_status == 0:
    self.physical_nic = cmd_result.stdout_text
```
But I don't think that's a good idea to mix up the definition of "physical_nic" this way, to make it can be either a fixed variant or an command.
`physical_nic = enp5s0f1`
`physical_nic = ip link | sed -n '/90:e2:ba:7d:33:3d/{g;1!p;};h'|awk -F': ' '{print $2}'|head -1`
Even for the command, to get it from a hard code mac address in the command is also not a good idea.

Reply:
Thanks, qizhu, I have update the codes.

it's very usefully in multiple ethernet machine. like 3 or more ethernet, it's not good to re-find the network name in different system(like rhel7. rhel8), if the network name could be find with mac address, we don't care about it and it provides more automated．

@luckyh, do you have any commands?

Reply:
Well, it makes sense to me that you want a dynamic way to get a physical nic, but I'd like to request you to explicitly point out that the value string is a shell command, e.g.
```
physical_nic = shell:ip link | sed ...
```
thus, you can exactly know whether you should call process.run
```python
if physical_nic.split(':', 1) == "shell":
    self.physical_nic = ...
```

Reply:
@luckyh , I have update the code. please help review. thanks